lib/AST/ASTPrinter.cpp

```diff
@@ -6387,10 +6387,27 @@ class TypePrinter : public TypeVisitor<TypePrinter> {
 
       // Print based on the type.
       Printer << "some ";
-      if (auto inheritedType = decl->getInherited().front().getType())
-        inheritedType->print(Printer, Options);
-      else
-        Printer << "Any";
+      if (!decl->getConformingProtocols().empty()) {
+        ArrayRef<ProtocolDecl *> protocols = decl->getConformingProtocols();
+        bool printed = false;
+        for (unsigned i = 0, e = protocols.size(); i < e; i++) {
+          if (auto printType = protocols[i]->getDeclaredType()) {
+            if (printed)
+              Printer << " & ";
+
+            printType->print(Printer, Options);
+
+            printed = true;
+          }
```

Comment:
If there's no declared type (which I think can't happen), you can print `protocols[i]->getName()` instead, so nothing will get dropped.